### PR TITLE
Unify helper isEmpty() in a single location

### PR DESCRIPTION
**Pull request by lucasponce**
While developing plugins/alerters I realized we have copies of same methods spread across several classes.
Unify them to reduce some noise, probably other similar aux methods can follow same strategy.
Anyway, it's just a minor cleaning PR.

### CODE REVIEW

File: api/src/main/java/org/hawkular/alerts/api/services/TriggersCriteria.java [hunk 1]
```diff
@@ -79,8 +81,7 @@ public boolean hasTagCriteria() {
     }
 
     public boolean hasTriggerIdCriteria() {
-        return null != triggerId
-                || (null != triggerIds && !triggerIds.isEmpty());
+        return null != triggerId || !isEmpty(triggerIds);
```
Comment (jshaughn) [inline api/src/main/java/org/hawkular/alerts/api/services/TriggersCriteria.java]: This actually could be
`return !isEmpty(triggerId) || !isEmpty(triggerIds);`

File: api/src/main/java/org/hawkular/alerts/api/services/ActionsCriteria.java [hunk 2]
```diff
@@ -164,19 +166,19 @@ public void setThin(boolean thin) {
     }
 
     public boolean hasActionIdCriteria() {
-        return null != actionId || (null != actionIds && !actionIds.isEmpty());
+        return null != actionId || !isEmpty(actionIds);
```
Comment (jshaughn) [inline api/src/main/java/org/hawkular/alerts/api/services/ActionsCriteria.java]: same change

File: api/src/main/java/org/hawkular/alerts/api/services/ActionsCriteria.java [hunk 3]
```diff
@@ -164,19 +166,19 @@ public void setThin(boolean thin) {
...
     }
 
     public boolean hasActionPluginCriteria() {
-        return null != actionPlugin || (null != actionPlugins && !actionPlugins.isEmpty());
+        return null != actionPlugin || !isEmpty(actionPlugins);
```
Comment (jshaughn) [inline api/src/main/java/org/hawkular/alerts/api/services/ActionsCriteria.java]: same change

File: api/src/main/java/org/hawkular/alerts/api/services/ActionsCriteria.java [hunk 4]
```diff
@@ -164,19 +166,19 @@ public void setThin(boolean thin) {
...
     }
 
     public boolean hasEventIdCriteria() {
-        return null != eventId || (null != eventIds && !eventIds.isEmpty());
+        return null != eventId || !isEmpty(eventIds);
```
Comment (jshaughn) [inline api/src/main/java/org/hawkular/alerts/api/services/ActionsCriteria.java]: same change

File: api/src/main/java/org/hawkular/alerts/api/services/ActionsCriteria.java [hunk 5]
```diff
@@ -164,19 +166,19 @@ public void setThin(boolean thin) {
...
     }
 
     public boolean hasResultCriteria() {
-        return null != result || (null != results && !results.isEmpty());
+        return null != result || !isEmpty(results);
```
Comment (jshaughn) [inline api/src/main/java/org/hawkular/alerts/api/services/ActionsCriteria.java]: same change

File: api/src/main/java/org/hawkular/alerts/api/services/AlertsCriteria.java [hunk 6]
```diff
@@ -293,31 +295,27 @@ public void setThin(boolean thin) {
     }
 
     public boolean hasAlertIdCriteria() {
-        return null != alertId
-                || (null != alertIds && !alertIds.isEmpty());
+        return null != alertId || !isEmpty(alertIds);
```
Comment (jshaughn) [inline api/src/main/java/org/hawkular/alerts/api/services/AlertsCriteria.java]: same change

File: api/src/main/java/org/hawkular/alerts/api/services/AlertsCriteria.java [hunk 7]
```diff
@@ -293,31 +295,27 @@ public void setThin(boolean thin) {
...
     }
 
     public boolean hasSeverityCriteria() {
-        return null != severity
-                || (null != severities && !severities.isEmpty());
+        return null != severity || !isEmpty(severities);
```
Comment (jshaughn) [inline api/src/main/java/org/hawkular/alerts/api/services/AlertsCriteria.java]: same change

File: api/src/main/java/org/hawkular/alerts/api/services/AlertsCriteria.java [hunk 8]
```diff
@@ -293,31 +295,27 @@ public void setThin(boolean thin) {
...
     }
 
     public boolean hasStatusCriteria() {
-        return null != status
-                || (null != statusSet && !statusSet.isEmpty());
+        return null != status || !isEmpty(statusSet);
```
Comment (jshaughn) [inline api/src/main/java/org/hawkular/alerts/api/services/AlertsCriteria.java]: same change

File: api/src/main/java/org/hawkular/alerts/api/services/AlertsCriteria.java [hunk 9]
```diff
@@ -293,31 +295,27 @@ public void setThin(boolean thin) {
...
     }
 
     public boolean hasCTimeCriteria() {
         return (null != startTime || null != endTime);
     }
 
     public boolean hasTriggerIdCriteria() {
-        return null != triggerId
-                || (null != triggerIds && !triggerIds.isEmpty());
+        return null != triggerId || !isEmpty(triggerIds);
```
Comment (jshaughn) [inline api/src/main/java/org/hawkular/alerts/api/services/AlertsCriteria.java]: same change

File: api/src/main/java/org/hawkular/alerts/api/services/ActionsCriteria.java (hunk 2 above)
Comment (lucasponce) [inline api/src/main/java/org/hawkular/alerts/api/services/ActionsCriteria.java]: +1

File: api/src/main/java/org/hawkular/alerts/api/services/ActionsCriteria.java (hunk 3 above)
Comment (lucasponce) [inline api/src/main/java/org/hawkular/alerts/api/services/ActionsCriteria.java]: +1

File: api/src/main/java/org/hawkular/alerts/api/services/ActionsCriteria.java (hunk 4 above)
Comment (lucasponce) [inline api/src/main/java/org/hawkular/alerts/api/services/ActionsCriteria.java]: +1

File: api/src/main/java/org/hawkular/alerts/api/services/ActionsCriteria.java (hunk 5 above)
Comment (lucasponce) [inline api/src/main/java/org/hawkular/alerts/api/services/ActionsCriteria.java]: +1

File: api/src/main/java/org/hawkular/alerts/api/services/AlertsCriteria.java (hunk 6 above)
Comment (lucasponce) [inline api/src/main/java/org/hawkular/alerts/api/services/AlertsCriteria.java]: +1

File: api/src/main/java/org/hawkular/alerts/api/services/AlertsCriteria.java (hunk 7 above)
Comment (lucasponce) [inline api/src/main/java/org/hawkular/alerts/api/services/AlertsCriteria.java]: -1
severity is a Severity object, so, I don't see value wrapping a severity != null inside a isEmpty() on this case.

File: api/src/main/java/org/hawkular/alerts/api/services/AlertsCriteria.java (hunk 8 above)
Comment (lucasponce) [inline api/src/main/java/org/hawkular/alerts/api/services/AlertsCriteria.java]: -1
Same case, status is an Status object, so isEmpty() is just wrapping the null validation and hiding this.

File: api/src/main/java/org/hawkular/alerts/api/services/AlertsCriteria.java (hunk 9 above)
Comment (lucasponce) [inline api/src/main/java/org/hawkular/alerts/api/services/AlertsCriteria.java]: +1

File: api/src/main/java/org/hawkular/alerts/api/services/AlertsCriteria.java (hunk 8 above)
Comment (jshaughn) [inline api/src/main/java/org/hawkular/alerts/api/services/AlertsCriteria.java]: Yes, my bad.

File: api/src/main/java/org/hawkular/alerts/api/services/AlertsCriteria.java (hunk 7 above)
Comment (jshaughn) [inline api/src/main/java/org/hawkular/alerts/api/services/AlertsCriteria.java]: Yes, my bad.

**Comment by lucasponce** [comment]
@jshaughn anything else to merge this one ? thx !

**Pull request merged**